shared/actions/notifications.js

```diff
@@ -81,9 +83,14 @@ function * _listenKBFSSaga (): SagaGenerator<any, any> {
   yield put(setupChatHandlers())
 }
 
-function * _onRecievedBadgeState (action: Constants.ReceivedBadgeState): SagaGenerator<any, any> {
-  const {conversations} = action.payload.badgeState
+function * _onReceivedBadgeState (action: Constants.ReceivedBadgeState): SagaGenerator<any, any> {
+  const {badgeState} = action.payload
+  const {conversations} = badgeState
   yield put(badgeAppForChat(conversations))
+  const newMessages = badgeStateToUnreadMessages(badgeState)
+  if (newMessages === 0) {
+    yield put(clearLocalPushNotifications)
```

Comment:
aka:       
when the reducer does
  s.set('mobileAppBadgeCount', navBadges.get(chatTab, 0))